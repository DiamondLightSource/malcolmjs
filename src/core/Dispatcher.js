/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
