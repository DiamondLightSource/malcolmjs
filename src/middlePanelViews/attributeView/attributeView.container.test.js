import React from 'react';
import { createShallow } from '@material-ui/core/test-utils'; // , createMount } from '@material-ui/core/test-utils';
import configureStore from 'redux-mock-store';
import {
  malcolmTypes,
  Widget,
} from '../../malcolmWidgets/attributeDetails/attributeSelector/attributeSelector.component';

import AttributeViewer from './attributeView.container';

describe('attributeView', () => {
  let shallow;
  // let mount;
  let mockStore;
  let state;

  beforeEach(() => {
    mockStore = configureStore();
    shallow = createShallow({ dive: true });
    // mount = createMount();
    state = {
      malcolm: {
        parentBlock: 'test1',
        mainAttribute: 'health',
        layout: {
          blocks: [],
        },
        blocks: {
          test1: {
            attributes: [
              {
                calculated: {
                  name: 'health',
                },
                raw: {
                  meta: {
                    tags: [Widget.TEXTUPDATE],
                    typeid: malcolmTypes.string,
                  },
                },
              },
            ],
          },
        },
        blockArchive: {
          test1: {
            attributes: [{}],
          },
        },
      },
      viewState: {
        openParentPanel: true,
      },
      router: {
        location: {
          pathname: '',
        },
      },
    };
  });

  it('renders 1st tab correctly', () => {
    const wrapper = shallow(
      <AttributeViewer
        store={mockStore(state)}
        attributeName="health"
        blockName="test1"
        widgetTag={Widget.TEXTUPDATE}
        typeid={malcolmTypes.string}
      />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('renders 2nd tab correctly', () => {
    const wrapper = shallow(
      <AttributeViewer
        store={mockStore(state)}
        attributeName="health"
        blockName="test1"
        widgetTag={Widget.TEXTUPDATE}
        typeid={malcolmTypes.string}
        defaultTab={1}
      />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
