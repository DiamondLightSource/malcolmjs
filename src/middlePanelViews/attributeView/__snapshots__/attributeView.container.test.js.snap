// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attributeView renders 1st tab correctly 1`] = `
<WithTheme(TabbedPanel)
  defaultTab={0}
  tabLabels={
    Array [
      "Table",
      "Plot",
    ]
  }
  useSwipeable={false}
>
  <ArchiveTable
    attribute={Object {}}
  />
  <LoadableComponent
    attribute={Object {}}
    deriveState={[Function]}
    doTick={true}
  />
</WithTheme(TabbedPanel)>
`;

exports[`attributeView renders 2nd tab correctly 1`] = `
<WithTheme(TabbedPanel)
  defaultTab={1}
  tabLabels={
    Array [
      "Table",
      "Plot",
    ]
  }
  useSwipeable={false}
>
  <ArchiveTable
    attribute={Object {}}
  />
  <LoadableComponent
    attribute={Object {}}
    deriveState={[Function]}
    doTick={true}
  />
</WithTheme(TabbedPanel)>
`;
